src/geode/georepresentation/msh_input.cpp

```diff
@@ -132,38 +134,64 @@ namespace
             return elementary_entity_id_;
         }
 
-        const std::vector< geode::index_t >& element_vertices_id()
+    protected:
+        geode::index_t& nb_vertices()
         {
-            return element_vertices_id_;
+            return nb_vertices_;
+        }
+
+        std::vector< geode::index_t >& vertex_ids()
+        {
+            return vertex_ids_;
+        }
+
+        void read_vertex_ids()
+        {
+            std::string id_info;
+            vertex_ids().resize( nb_vertices() );
+            for( auto n : geode::Range{ nb_vertices() } )
+            {
+                iss_vertices_ids_ >> id_info;
```

Comment:
operator>> has a lot of overloads. You could create `geode::index_t id_info` and it will directly convert the string to the corresponding type. You can do this a different locations in this file.

Reply:
Ok, I did not know! Excellent!

Reply:
Why not using directly an index_t Instead of the string and the stoi ?